File: Cocktail/src/main/kotlin/org/programmers/cocktail/search/service/CommentsMapper.java
package org.programmers.cocktail.search.service;

import java.util.List;
import java.util.stream.Collectors;
import org.modelmapper.ModelMapper;
import org.modelmapper.PropertyMap;
import org.programmers.cocktail.entity.Comments;
import org.programmers.cocktail.search.dto.CommentsTO;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class CommentsMapper {
    private ModelMapper modelMapper;

    @Autowired
    public CommentsMapper(ModelMapper modelMapper) {
        this.modelMapper = modelMapper;

        // Comments -> CommentsTO 변환 매핑
        modelMapper.addMappings(new PropertyMap<Comments, CommentsTO>() {
            @Override
            protected void configure() {
                map().setUserId(source.getUsers().getId());
                map().setCocktailId(source.getCocktails().getId());
                map().setUserName(source.getUsers().getName());
            }
        });

        // CommentsTO -> Comments 변환 매핑
        modelMapper.addMappings(new PropertyMap<CommentsTO, Comments>() {
            @Override
            protected void configure() {
                map().getUsers().setId(source.getUserId());
                map().getCocktails().setId(source.getCocktailId());
            }
        });
    }

    public List<CommentsTO> convertToCommentsTOList(List<Comments> commentsList) {
        List<CommentsTO> commentsTOList = commentsList.stream()
            .map(comments -> modelMapper.map(comments, CommentsTO.class))
            .collect(Collectors.toList());
        return commentsTOList;
    }

    public Comments convertToComments(CommentsTO commentsTO) {
        return modelMapper.map(commentsTO, Comments.class);
    }
}

File: Cocktail/src/main/kotlin/org/programmers/cocktail/search/service/CommentsMapper.kt
package org.programmers.cocktail.search.service

import org.programmers.cocktail.entity.Comments
import org.programmers.cocktail.search.dto.CommentsTO
import org.springframework.stereotype.Service

@Service
object CommentsMapper {

    // Comments -> CommentsTO 변환
    fun convertToCommentsTO(comments: Comments?): CommentsTO {
        return CommentsTO(
            id = comments?.id,
            userId = comments?.users?.id,        // users 객체에서 id를 가져와서 userId로 설정
            cocktailId = comments?.cocktails?.id, // cocktails 객체에서 id를 가져와서 cocktailId로 설정
            content = comments?.content ?: throw RuntimeException("[Mapping Failure] content is null"),
            userName = comments.users.name,
            createdAt = comments.createdAt,
            updatedAt = comments.updatedAt
        )
    }

    // CommentsTO -> Comments 변환
    fun convertToComments(commentsTO: CommentsTO?): Comments {
        val comments = Comments()

        // userId와 cocktailId로 users, cocktails 객체를 설정
        comments.users.id = commentsTO?.userId
        comments.cocktails.id = commentsTO?.cocktailId
        comments.content = commentsTO?.content ?: throw RuntimeException("[Mapping Failure] content is null")

        return comments
    }

    // Comments 리스트 -> CommentsTO 리스트 변환
    fun convertToCommentsTOList(commentsList: List<Comments?>): List<CommentsTO> {
        return commentsList.map { convertToCommentsTO(it) }
    }
}